app/models/user.rb

```diff
@@ -7,4 +7,15 @@ class User < ActiveRecord::Base
   # :confirmable, :lockable, :timeoutable and :omniauthable
   devise :database_authenticatable, :registerable,
          :recoverable, :rememberable, :trackable, :validatable
+
+
+  def can_edit?(resource)
+    admin || owns?(resource)
+  end
+
+  private
+
+  def owns?(resource)
+    resource ? (self == resource.user) : false
```

Comment:
That will break if resource is nil (like if 'resource' is the result of a `.find` call that returns nil)